### PR TITLE
Update Docs

### PR DESCRIPTION
**Pull request by felsweg-iota**
# Description of change

This PR extends the welcome docs to Stronghold, give a more clear overview on the intended functionality of Stronghold

## Links to any relevant issues

None.

## Type of change

- [x] Documentation Fix

## How the change has been tested

No additional tests have been provided 

## Change checklist
- [x] I have followed the contribution guidelines for this project
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by vuongDang — APPROVED** [review]

File: documentation/docs/welcome.md [hunk 1]
```diff
@@ -24,6 +24,12 @@ protect any digital secret.
 Stronghold is a secure database for working with cryptography, which ensures that secrets (like private keys) are never
 revealed, but can be used according to best practices.
 
+Stronghold can be seen as some kind of isolated container for secrets that may be either a private key, or some other bytes of data, that should never be exposed. In order to work with Stronghold one uses  so called procedures to either generate new keys, store data into the vault, derive keys or work with the data in place. The difference to a "traditional" password store is, that secret data is never accessed directly but can be worked with the aforementioned procedures. Whenever there is a need to sign some data with a private key, one would be required to call a respective function to work with it.
```
Comment (lucas-tortora) [inline documentation/docs/welcome.md]:
```suggestion
You can think of Stronghold as an isolated container for secrets. The secret can be a private key or other bytes of data that should never be exposed. You can interact with Stronghold using procedures to [generate new keys](how_tos/cli/generate_key_pair), [store data in the vault](how_tos/cli/store_read_write), [derive keys](how_tos/cli/derive_slip_10), or work with the data. The difference to a "traditional" password store is that you can never access the secret data directly. You will need to use procedures. For example, if you need to sign data with your private key, you must call a procedure.
```

File: documentation/docs/welcome.md [hunk 2]
```diff
@@ -24,6 +24,12 @@ protect any digital secret.
...
 
+Stronghold can be seen as some kind of isolated container for secrets that may be either a private key, or some other bytes of data, that should never be exposed. In order to work with Stronghold one uses  so called procedures to either generate new keys, store data into the vault, derive keys or work with the data in place. The difference to a "traditional" password store is, that secret data is never accessed directly but can be worked with the aforementioned procedures. Whenever there is a need to sign some data with a private key, one would be required to call a respective function to work with it. 
+
+Generally speaking there are two states where the secret data remains. The runtime operation makes use of Clients. Clients can be thought of as a context-based secure data container with all the functionality to work with sensitive data. In order to persist the runtime data, there are the Snapshot facilities. The Snapshot is actually twofold. At the lowest level, the Snapshot is a highly encrypted file which is being represented by an in-memory instance of Snapshot itself, which itself is encrypted and protected via the same means as the Vault.
```
Comment (lucas-tortora) [inline documentation/docs/welcome.md]:
```suggestion
Generally speaking, there are two states where the secret data remains, [Clients](reference/structure/client) and the [Snapshot](reference/structure/engine/snapshot) facilities.

The runtime operation uses Clients. You can think of Clients as a context-based secure data container that includes all the functionality you need to work with sensitive data. 

Stronghold uses Snapshot facilities to persist the runtime. The Snapshot is actually twofold. At the lowest level, the Snapshot is a highly encrypted file represented by an in-memory instance of Snapshot itself, which is encrypted and protected via the same means as the Vault.
```

File: documentation/docs/welcome.md [hunk 3]
```diff
@@ -24,6 +24,12 @@ protect any digital secret.
...
+
+Additionally to the secure `Vault` type, Stronghold provides an eviciting cache to store non-sensitive data called the `Store`. Use the `Store` to keep session based configuration data.
```
Comment (lucas-tortora) [inline documentation/docs/welcome.md]:
```suggestion
Additionally to the secure secret database [Vault](reference/structure/engine/vault), Stronghold provides an evicting cache to store non-sensitive data called the [Store](reference/structure/engine/store). Use the Store to keep session-based configuration data. 
```

**Review by lucas-tortora — COMMENTED** [review]
I've added a few links and made some minor style and structural changes. Awesome contribution :)

File: documentation/docs/welcome.md (hunk 1 above)
Comment (felsweg-iota) [reply documentation/docs/welcome.md]: I was using "you" before and changed it to be more neutral :sweat_smile: But I do see the point.

**Comment by felsweg-iota** [comment]
Thanks @lucas-tortora !

**Review by lucas-tortora — APPROVED** [review]

**Pull request merged**